dspace-server-webapp/src/test/java/org/dspace/app/rest/BitstreamRestControllerIT.java

```diff
@@ -425,5 +514,127 @@ private int getNumberOfPdfPages(byte[] content) throws IOException {
+
+        getClient(getAuthToken(eperson.getEmail(), password))
+                .perform(get("/api/core/bitstreams/" + bitstream.getID() + "/format"))
+                .andExpect(status().isOk());
+
+        getClient(getAuthToken(admin.getEmail(), password))
+                .perform(get("/api/core/bitstreams/" + bitstream.getID() + "/format"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", matchBitstreamFormat(
+                        supportedFormat.getID(),
+                        supportedFormat.getMIMEType(),
+                        supportedFormat.getDescription(),
+                        supportedFormat.getShortDescription(),
+                        "SUPPORTED"
+                )));
+    }
+
+    @Test
+    public void updateBitstreamFormat() throws Exception {
+
+        getClient().perform(
+                put("/api/core/bitstreams/" + bitstream.getID() + "/format")
+                        .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                        .content(
+                                "https://localhost:8080/spring-rest/api/core/bitstreamformat/" + knownFormat.getID()
+                        )
+        ).andExpect(status().isUnauthorized());
+
+        getClient(getAuthToken(eperson.getEmail(), password)).perform(
+                put("/api/core/bitstreams/" + bitstream.getID() + "/format")
+                        .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                        .content(
+                                "https://localhost:8080/spring-rest/api/core/bitstreamformat/" + knownFormat.getID()
+                        )
+        ).andExpect(status().isForbidden());
+
+        context.turnOffAuthorisationSystem();
+
+        resourcePolicy = resourcePolicyService.create(context);
+        resourcePolicy.setEPerson(eperson);
+        resourcePolicy.setAction(WRITE);
+        resourcePolicy.setdSpaceObject(bitstream);
+
+        bitstream = context.reloadEntity(bitstream);
+        bitstreamService.addMetadata(context, bitstream, "dc", "format", null, null, "test format");
+
+        context.commit();
+
+        getClient(getAuthToken(eperson.getEmail(), password)).perform(
+                put("/api/core/bitstreams/" + bitstream.getID() + "/format")
+                        .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                        .content(
+                                "https://localhost:8080/spring-rest/api/core/bitstreamformat/" + knownFormat.getID()
+                        )
+        ).andExpect(status().isOk());
+
+        assertThat(knownFormat, equalTo(context.reloadEntity(bitstream).getFormat(context)));
+        assertTrue(isEmpty(
+                bitstreamService.getMetadataByMetadataString(context.reloadEntity(bitstream), "dc.format")
+        ));
+
+        bitstream = context.reloadEntity(bitstream);
+        bitstreamService.addMetadata(context, bitstream, "dc", "format", null, null, "another test format");
```

Comment:
I'd recommend creating two bitstreams here, again using a `builder` instead of changing the value of `dc.format` in the same bitstream. That will save you from having to do all `context.reloadEntity(bitstream)` calls (which are all over this method).